app/main.py

```diff
@@ -1,3 +1,71 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", float]) -> "Distance":
```

Comment:
Do not use `Union` and do not wrap `Distance` in quotes. Besides that, indicate that `other` could `int` as well: 
```suggestion
    def __add__(self, other: Distance | float | int) -> Distance:
```
Apply this change to all type annotations where applicable. In case you face errors related to the type annotation, add `from __future__ import annotations`.